lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -60,15 +66,11 @@ def query_api(
 
         return query_api
 
-    def token(self, client, config: PbenchServerConfig, user: str) -> str:
-        response = client.post(
-            f"{config.rest_uri}/login",
-            json={"username": user, "password": "12345"},
-        )
-        assert response.status_code == HTTPStatus.OK
-        data = response.json
-        assert data["auth_token"]
-        return data["auth_token"]
+    def token(self, user: str) -> str:
+        roles = None
+        if user == "test_admin":
+            roles = ["ADMIN"]
+        return generate_token(username=user, pbench_client_roles=roles)
```

Comment:
If you don't just delete this code in favor of references to the `pbench_token` or `pbench_admin_token` fixtures, this looks like a nice spot for a ternary.  😁

Reply:
Having seen the resulting mess in other contexts, "ugh". (Granted, this is probably less indented than some...)